src/darwin/Framework/CHIPTests/MTRDeviceTests.m

```diff
@@ -2605,6 +2606,166 @@ - (void)test028_TimeZoneAndDST
 #endif // MTR_ENABLE_PROVISIONAL
 }
 
+/**
+ * Given a path relative to the Matter root, create an absolute path to the file.
+ */
+- (NSString *)absolutePathFor:(NSString *)matterRootRelativePath
+{
+    // Find the right absolute path to our file.  PWD should
+    // point to our src/darwin/Framework.
+    NSString * pwd = [[NSProcessInfo processInfo] environment][@"PWD"];
+    NSMutableArray * pathComponents = [[NSMutableArray alloc] init];
+    [pathComponents addObject:[pwd substringToIndex:(pwd.length - @"src/darwin/Framework".length)]];
+    [pathComponents addObjectsFromArray:[matterRootRelativePath pathComponents]];
+    return [NSString pathWithComponents:pathComponents];
+}
+
+/**
+ * Create a task given a path relative to the Matter root.
+ */
+- (NSTask *)createTaskForPath:(NSString *)path
+{
+    NSTask * task = [[NSTask alloc] init];
+    [task setLaunchPath:[self absolutePathFor:path]];
+    return task;
+}
+
+- (NSError *)generateLogFile:(NSString *)outFile
+                        size:(unsigned long long)size
+{
+    // Remove the file if one exists
+    [[NSFileManager defaultManager] removeItemAtPath:outFile error:nil];
+
+    // Create the file
+    [[NSFileManager defaultManager] createFileAtPath:outFile contents:nil attributes:nil];
+
+    NSString * content = @"The quick brown fox jumps over the lazy dog";
+    NSFileHandle * handle = [NSFileHandle fileHandleForWritingAtPath:outFile];
+    if (handle == nil)
+    {
+        NSLog(@"Failed to generate the log file %@", outFile);
```

Comment:
But then we go on to use handle?

Reply:
i changed this yesterday. looks like i didn't check the code thoroughly at all

Reply:
fixed.